features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioPalSecurityManager.cpp

```diff
@@ -0,0 +1,601 @@
+                    connection,
+                    (pairing_failure_t::type) status
+                );
+            } else if (status == SMP_ERR_MEMORY) {
+                // note: report the error as an unspecified error
+                handler->on_pairing_error(
+                    connection,
+                    pairing_failure_t::UNSPECIFIED_REASON
+                );
+            } else if (msg->status == SMP_ERR_TIMEOUT) {
+                handler->on_pairing_timed_out(connection);
+            } else {
+                // note: report the error as an unspecified error
+                handler->on_pairing_error(
+                    connection,
+                    pairing_failure_t::UNSPECIFIED_REASON
+                );
+            }
+            return true;
+        }
+
+        case DM_SEC_ENCRYPT_IND: {
+            dmSecEncryptIndEvt_t* evt = (dmSecEncryptIndEvt_t*) msg;
+            // note: the field usingLtk of the message indicates if an LTK was
+            // used to encrypt the link
+            handler->on_link_encryption_result(evt->hdr.param, link_encryption_t::ENCRYPTED);
+            return true;
+        }
+
+        case DM_SEC_ENCRYPT_FAIL_IND: {
+            // note: msg->status contains the encryption failure status
+            handler->on_link_encryption_result(msg->param, link_encryption_t::NOT_ENCRYPTED);
+            return true;
+        }
+
+        case DM_SEC_AUTH_REQ_IND: {
+            dmSecAuthReqIndEvt_t* evt = (dmSecAuthReqIndEvt_t*) msg;
+            connection_handle_t connection = evt->hdr.param;
+
+            if (evt->oob) {
+                handler->on_legacy_pairing_oob_request(connection);
+            } else if (evt->display) {
+                if (get_security_manager()._use_default_passkey) {
+                    handler->on_passkey_display(
+                        connection,
+                        get_security_manager()._default_passkey
+                    );
+                    DmSecAuthRsp(
+                        connection,
+                        3,
+                        reinterpret_cast<uint8_t*>(&(get_security_manager()._default_passkey))
+                    );
+                } else {
+                    // FIXME: generate a random passkey
+                    passkey_num_t passkey = 0x00654321;
+                    handler->on_passkey_display(connection, passkey);
+                    DmSecAuthRsp(
+                        connection, 3, reinterpret_cast<uint8_t*>(&passkey)
+                    );
+                }
+            } else {
+                handler->on_passkey_request(connection);
+            }
+            return true;
+        }
+
+        case DM_SEC_KEY_IND: {
+            dmSecKeyIndEvt_t* evt = (dmSecKeyIndEvt_t*) msg;
+            connection_handle_t connection = evt->hdr.param;
+
+            switch(evt->type) {
+                case DM_KEY_LOCAL_LTK:
+                    // FIXME: forward local ltk
+#if 0
```

Comment:
This block should be removed and the local key should be passed to the handler.

Reply:
is this a self comment? I don't know what you mean